lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs

```diff
@@ -261,11 +265,15 @@ canonicalRequest meth path digest query chs shs =
         toBS digest
       ]
 
-escapedPath :: Request a -> Path
-escapedPath r = Tag . toBS . escapePath $
-  case _serviceAbbrev (_requestService r) of
-    "S3" -> _requestPath r
-    _ -> collapsePath (_requestPath r)
+escapedPaths :: Request a -> (Path, CanonicalPath)
```

Comment:
Can we return `CanonicalPath` only, and not a tuple?

Reply:
Thinking about this more, yes it's possible. Where the actual http request is constructed, we'd need to do an unescape: https://github.com/brendanhay/amazonka/blob/9043cb4dc894a1a0d0f24375d70df5cb5e3b1ac8/lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs#L160  But it'd have to be typed and not ByteString since it has to know how to go back from twice escaped to only once-escaped as the http request is always made against the once-escaped URL.

Reply:
Okay, I didn't realise that the left hand side of the tuple was still used. Would you mind splitting into two functions `escapePath` and `escapeCanonicalPath`?

Reply:
Yep.